x-pack/platform/plugins/shared/encrypted_saved_objects/integration_tests/ci_checks/check_registered_types.test.ts

```diff
@@ -0,0 +1,324 @@
+        "synthetics-param": "747ba9d1b7addf5b131713abe7868bd767af6ce0cf8b6b0f335f4ef34b280c7e",
+        "task": "2d8e9bf532f469805b82051f545b915785d99eabfa050cb1aefbc715c6096b97",
+        "uptime-synthetics-api-key": "5ca81f180763e85397fa8c6508adcd60efd0f916e29bac6dcd5b4564f1db7375",
+      }
+    `);
+    expect(Object.keys(hashMap).length).toEqual(ESO_TYPES_COUNT);
+  });
+
+  // This test is meant to fail and require an update when a new model version is introduced to any ESO types.
+  // There are tests in core which will catch when an SO type is changed, and help the Core team enforce when
+  // a model version needs to be added. This purpose of this test is to ensure that new model versions for ESOs
+  // are implmented to the security team's guidelines to adhere to all zero-downtime upgrade considerations.
+  it('detecting new model versions in registered encrypted types', () => {
+    const esoTypes = esoService.getRegisteredTypes();
+    const soTypestoCheck = typeRegistry
+      .getAllTypes()
+      .filter((soType) => esoTypes.includes(soType.name));
+
+    const modelVersionMap: string[] = sortBy(soTypestoCheck, 'name').flatMap((type, index) => {
+      const typeMigrationInfo = sortBy(extractMigrationInfo(type).modelVersions, 'version')
+        .reverse()
+        .map((mv) => `${type.name}|${mv.version}`);
+
+      return typeMigrationInfo;
+    });
+
+    expect(modelVersionMap).toMatchInlineSnapshot(`
+      Array [
+        "action|1",
+        "action_task_params|2",
+        "action_task_params|1",
+        "ad_hoc_run_params|2",
+        "ad_hoc_run_params|1",
+        "alert|6",
+        "alert|5",
+        "alert|4",
+        "alert|3",
+        "alert|2",
+        "alert|1",
+        "api_key_pending_invalidation|1",
+        "connector_token|1",
+        "fleet-fleet-server-host|2",
+        "fleet-fleet-server-host|1",
+        "fleet-uninstall-tokens|1",
+        "ingest-download-sources|1",
+        "ingest-outputs|8",
+        "ingest-outputs|7",
+        "ingest-outputs|6",
+        "ingest-outputs|5",
+        "ingest-outputs|4",
+        "ingest-outputs|3",
+        "ingest-outputs|2",
+        "ingest-outputs|1",
+        "synthetics-monitor|2",
+        "synthetics-monitor|1",
+        "task|6",
+        "task|5",
+        "task|4",
+        "task|3",
+        "task|2",
+        "task|1",
+      ]
+    `);
+  });
+
+  // This test is meant to more granularly detect what changes when occured to an ESO's migration-related properties.
+  // Though there is a similar test in core, this is relevent to ESOs and the Kibana Security team.
+  // Just update the snapshot by running this test file via jest_integration with `-u` and push the update.
+  // The intent is to trigger a code review from the Kibana Security team to review the ESO changes.
+  // The number of types in the hashMap should never be reduced, it should only increase.
+  // it('detecting modelVersion and schema changes in registered types', () => {
```

Comment:
I think we can remove this test as redundant with Core's test, and instead look for better, more ESO-specific ways to improve our test suite.